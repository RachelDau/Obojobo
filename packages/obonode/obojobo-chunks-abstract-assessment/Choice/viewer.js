import adapter from './adapter'
import Common from 'obojobo-document-engine/src/scripts/common'
import ViewerComponent from './viewer-component'

Common.Registry.registerModel('ObojoboDraft.Chunks.AbstractAssessment.Choice', {
	adapter: adapter,
	componentClass: ViewerComponent,
	type: 'chunk'
})
