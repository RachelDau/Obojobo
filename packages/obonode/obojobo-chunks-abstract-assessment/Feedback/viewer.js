import Common from 'obojobo-document-engine/src/scripts/common'
import ViewerComponent from './viewer-component'

Common.Registry.registerModel('ObojoboDraft.Chunks.AbstractAssessment.Feedback', {
	adapter: null,
	componentClass: ViewerComponent,
	type: 'chunk'
})
