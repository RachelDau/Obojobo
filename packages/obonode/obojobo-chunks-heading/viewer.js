import adapter from './adapter'
import Common from 'obojobo-document-engine/src/scripts/common'
import ViewerComponent from './viewer-component'

Common.Registry.registerModel('ObojoboDraft.Chunks.Heading', {
	adapter: adapter,
	componentClass: ViewerComponent,
	type: 'chunk',
	getNavItem(model) {
		switch (model.modelState.headingLevel) {
			// when 1
			// 	type: 'link',
			// 	label: model.modelState.textGroup.first.text.value,
			// 	path: [model.modelState.textGroup.first.text.value.toLowerCase().replace(/ /g, '-')],
			// 	showChildren: false

			case 1:
			case 2:
				if (model.modelState.headingLevel === 1 && model.getIndex() === 0) {
					return null
				}

				return {
					type: 'sub-link',
					label: model.modelState.textGroup.first.text,
					path: [
						model
							.toText()
							.toLowerCase()
							.replace(/ /g, '-')
					],
					showChildren: false
				}

			default:
				return null
		}
	}
})
