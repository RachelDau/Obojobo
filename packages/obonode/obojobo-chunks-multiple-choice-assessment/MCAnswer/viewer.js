import Common from 'obojobo-document-engine/src/scripts/common'
import ViewerComponent from './viewer-component'

Common.Registry.registerModel('ObojoboDraft.Chunks.MCAssessment.MCAnswer', {
	adapter: null,
	componentClass: ViewerComponent,
	type: 'chunk'
})
