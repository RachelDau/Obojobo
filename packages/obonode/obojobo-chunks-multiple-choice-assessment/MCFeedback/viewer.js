import Common from 'obojobo-document-engine/src/scripts/common'
import ViewerComponent from './viewer-component'

Common.Registry.registerModel('ObojoboDraft.Chunks.MCAssessment.MCFeedback', {
	adapter: null,
	componentClass: ViewerComponent,
	type: 'chunk'
})
