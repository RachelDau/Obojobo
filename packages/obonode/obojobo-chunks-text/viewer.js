import Common from 'obojobo-document-engine/src/scripts/common'
import ViewerComponent from './viewer-component'

Common.Registry.registerModel('ObojoboDraft.Chunks.Text', {
	adapter: Common.chunk.textChunk.TextGroupAdapter,
	componentClass: ViewerComponent,
	default: true,
	type: 'chunk'
})
