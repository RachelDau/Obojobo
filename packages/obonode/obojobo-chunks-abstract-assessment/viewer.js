// Any component that inherits AbstractAssessment should define their own
// Answer subcomponent
// import './Answer/viewer'
import './Choice/viewer'
import './Feedback/viewer'

// Note: AbstractAssessment does not Directly register a model
// to the Common.Registry; inheriting components that implement 
// specific behaviors are expected to register themselves. 