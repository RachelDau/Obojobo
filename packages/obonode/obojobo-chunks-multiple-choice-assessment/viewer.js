import adapter from './adapter'
import Common from 'Common'
import ViewerComponent from './viewer-component'

import './MCAnswer/viewer'
import './MCChoice/viewer'
import './MCFeedback/viewer'

Common.Registry.registerModel('ObojoboDraft.Chunks.MCAssessment', {
	adapter: adapter,
	componentClass: ViewerComponent,
	type: 'chunk'
})
