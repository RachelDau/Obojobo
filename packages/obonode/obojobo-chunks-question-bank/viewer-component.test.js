import React from 'react'
import renderer from 'react-test-renderer'

import QuestionBank from './viewer-component'
import OboModel from 'obojobo-document-engine/src/scripts/common/models/obo-model'

require('./viewer') // used to register this oboModel

describe('QuestionBank', () => {
	test('QuestionBank component', () => {
		const model = OboModel.create({
			id: 'id',
			type: 'ObojoboDraft.Chunks.QuestionBank',
			children: [
				{
					id: 'child-id',
					type: 'ObojoboDraft.Chunks.QuestionBank'
				}
			]
		})

		const moduleData = {
			focusState: {}
		}

		const component = renderer.create(<QuestionBank model={model} moduleData={moduleData} />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})
})
