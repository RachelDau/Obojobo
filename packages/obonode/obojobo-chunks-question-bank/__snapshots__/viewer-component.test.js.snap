// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionBank QuestionBank component 1`] = `
<div
  className="component obojobo-draft--chunks--question-bank"
  data-focussed={false}
  data-id="id"
  data-obo-component={true}
  data-type="ObojoboDraft.Chunks.QuestionBank"
  id="obo-id"
  tabIndex={-1}
>
  <div
    className="component obojobo-draft--chunks--question-bank"
    data-focussed={false}
    data-id="child-id"
    data-obo-component={true}
    data-type="ObojoboDraft.Chunks.QuestionBank"
    id="obo-child-id"
    tabIndex={-1}
  />
</div>
`;
